open_instruct/finetune.py

```diff
@@ -1065,7 +1043,7 @@ def main(args: FlatArguments):
             json.dump(metadata_blob, f)
 
         # upload metadata to the dataset if set
-        if args.hf_metadata_dataset:
+        if args.push_to_hub and args.hf_metadata_dataset:
```

Comment:
Odd change? Only upload metadata if we push to HF?

Reply:
Oh good catch. After checking what metadata is uploaded, I agree this shouldn't be added here.

I added there as this if branch (see below) uses the `args.hf_repo_revision`, which only exists when `push_to_hub` is set to be true. What would be a good argument to use for `hf_dataset_save_dir`?
```
        # upload metadata to the dataset if set
        if args.hf_metadata_dataset:
            upload_metadata_to_hf(
                metadata_blob,
                "metadata.json",
                args.hf_metadata_dataset,
                "results/" + args.hf_repo_revision,  # to match what the auto-evals name as.
            )
```

Reply:
Good question.... it should match however downstream chooses the name, which I'm not sure about on the top of my head.